internal/graph/cache_key_hasher.go

```diff
@@ -0,0 +1,84 @@
+package graph
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/cespare/xxhash/v2"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+)
+
+type hasher interface {
+	WriteString(value string) error
+}
+
+// CacheKeyHasher implements a key hash using Hash64 for computing cache keys in a stable way.
+type CacheKeyHasher struct {
```

Comment:
No need for this to be publicly exported from the containing package.

https://github.com/openfga/openfga/pull/1038/files#diff-7dedee4631643575bf44454c0194c373e5f67dde51a28b4a4ce4b816380e48dbR6-R12

Reply:
I get this warning if I don't export the struct: 
<img width="527" alt="image" src="https://github.com/openfga/openfga/assets/5374887/aa412dc5-3797-4a52-a0dd-d2e949a5e39b">

Reply:
It works fine on my end if you move the relevant symbols into `internal/keys` and unexport the publicly exported symbols in favor of the constructors. I'd advise we reintroduce the changes that were originally included in `feat/abac`, because these changes made are honestly just scope creep at the moment.

The hasher and hashable value interfaces exist in their own right without the `graph` package and should be separated as such. Composing hashers together should also be possible, such is the case with the `tupleKeysHasher` and the `cacheKeyHasher`. When `feat/abac` is merged we introduce other hashable values that we can append as another form of a hasher to the hash chain. Point is, different kinds of keys can be hashed using different hashable values. The interface is simple and composable and exhibits the use case.

Reply:
Hmm ok. For now it works but the moment we try to use this code from another package, it won't

Reply:
I'm not sure I understand what you mean. This code already being used outside of the package that defines it and it works fine.

`internal/graph/cached_resolver.go` uses the constructor to construct a `cacheKeyHasher` and then subsequently uses it.